### ISSUE TITLE
browser instance duplicates at all workspaces

### ISSUE THREAD

**Issue by gucio321** [brave/brave-browser#33498]
## Description 

Hi,
Well its the strangest bug I've ever seen and I just don't know what to do with it so forgive if wrong repo :smile:
When I open a brave on my fedora linux with gnome shell it appears on EVERY workspace. And it seems to be the same window (doing any changes in an instance on one workspace changes things in other).
![image](https://github.com/brave/brave-browser/assets/73652197/47314373-1383-4c50-9e7e-5d8c2094869f)

## Steps to Reproduce

   1. just open brave

## Actual result:

new window will duplicate at all workspaces
## Expected result:
it should open one window in current workspac

## Reproduces how often: 

on my system I can reproduce it every time I run my browser

## Brave version (brave://version info)

<details><summary>Details</summary>
<p>

```console
Brave	1.58.137 Chromium: 117.0.5938.153 (Official Build) (64-bit) 
Revision	40b532ccf3bc15f40c6165989b143955260b872c
OS	Linux
JavaScript	V8 11.7.439.21
User Agent	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36
Command Line	/opt/brave.com/brave/brave --disable-domain-reliability --enable-dom-distiller --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --flag-switches-begin --flag-switches-end --component-updater=url-source=https://go-updater.brave.com/extensions --disable-nacl
Executable Path	/opt/brave.com/brave/brave
Profile Path	/home/mszeptuch/.config/BraveSoftware/Brave-Browser/Default
Active Variations	AllowCertainClientHintsStudy:Enabled
BraveAdblockMobileNotificationsListDefault:Enabled
BraveAds.AdServingStudy:DefaultAdNotificationsPerHour=10/MaximumAdNotificationsPerDay=100/MaximumInlineContentAdsPerHour=6/MaximumInlineContentAdsPerDay=20/AdServingVersion=2
BraveAutoTranslateStudy:AutoTranslateDisabled
BraveDebounceStudy:Enabled
BraveForgetFirstPartyStorage:Enabled
BraveGoogleSignInPermissionStudy:Disabled
BraveHttpsByDefaultRolloutStudy:Enabled
BraveP3AConstellationReleaseRollout:Enabled
BraveRewardsWebUiPanelStudy:Enabled
BraveScreenFingerprintingBlockerStudy:Enabled
BraveSearchAdStudy:Enabled
ClampPlatformVersionClientHint:Disabled
CookieListDefaultStudy:Enabled
Default1pBlockingStudy:Disabled
DisableReduceLanguage:Enabled
EphemeralStorageStudy:Enabled
FirstPartyEphemeralDomainBlockStudy:Enabled
ForceDisableBlinkFeatures:Enabled
PartitionBlinkMemoryCacheStudy:Enabled
PartitionConnectionsByNetworkIsolationKeyStudy:Enabled
PermissionLifetimeReleaseStudy:Enabled
PrettyPrintJSONDocument:Disabled_EmergencyKillSwitch
SpeedreaderReleaseStudy:Enabled
UserActivityStudy:Triggers
WhatsNewStudy:Enabled
```

</p>
</details> 

## Version/Channel Information:

Can't say... I use version from `dnf`

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? I think no...
- Is the issue reproducible on the latest version of Chrome? I've tried chromium 117.0.5938.132 Fedora Project and it works normally

## Miscellaneous Information:

- Fedora release 38 (Thirty Eight)
- GNOME Shell 44.5

**Comment by gucio321** [comment]
ok... THe issue resolved itself after closing all the tabs (and then whole browser) and re-opening it.
I think I'll close it as the bug is fixed for me and I can't reproduce it back.

**Comment by gucio321** [comment]
well, it abbeard again.

**Comment by lachlan-bailey** [comment]
I would like to share I have also been experiencing this issue.

Something I have noticed is this issue appears when I unplug my laptop from my external display.

## Brave Version (brave://version)

1.61.109 Chromium: 120.0.6099.144 (Official Build) (64-bit) 

## Misc. Information

**Operating System:** Ubuntu 22.04.3 LTS

**GNOME Version:** 42.9

**Windowing System:** X11

**Comment by bjjblackbelt** [comment]
Happens to me too.

Brave version: 1.64.109
OS: Fedora 39
Gnome: 45.5
Windowing System: Wayland
Kernel: 6.7.9-200

**Comment by SeanPhilippi** [comment]
I'm seeing this too. Have had this for many months. I only notice it when not plugged into external monitors. Will check later if it happens upon unplugging from them. Also only the instances of Brave that were not on my primary screen get duplicated. Or at least my window with less tabs always get duplicated, but not my screen with the most tabs. Closing these windows and re-opening using history does not fix it, it just duplicates across workspaces again. 

Brave version: 1.65.89
OS: Ubuntu 23.10
Gnome: 45.2
Windowing System: X11
Kernel: 6.5.0-26-generic

**Comment by gucio321** [comment]
generally I completely closed my brave and now I didn't see this bug since that.
I'm on fedora+wayland+gnome and using extenral monitor

**Comment by SeanPhilippi** [comment]
@gucio321 I tried closing Brave completely, and I just plugged into my external monitors, plugged back in, unplugged. Seems fixed! Thanks!

**Comment by Rick1029** [comment]
I don't believe this is a brave specific issue as it just happened to me on PopOS 22.04 running Firefox. It's either a Browser issue as a whole or more likely something to do with Gnome, however it is strange that it happens on Fedora and PopOS given that PopOS runs its own Cosmic desktop on top of Gnome.

![image](https://github.com/brave/brave-browser/assets/62437512/a5697e22-8bb2-4396-ac8b-491b02d180c0)
As you can see, this github page where I am reporting is duplicated on every workspace. I can start typing in my top workspace and finish exactly where I left off, with my cursor still blinking, in another, which insinuates it is exactly the same window and not just duplicates. Editing one workspace makes the same changes on another (to do with this browser window only) Closing the window stops the effect for that window and closes it on every workspace, opening new windows doesn't necessarily mean the effect returns. Creating a new workspace (moving some application, regardless of whether its Firefox or not, to another workspace, duplicates the already duplicated windows. The new application is unaffected.)

**Comment by jorgebotas** [comment]
I have the same issue Ubuntu 24. Plugged and unplugged from external monitors. Any work around ?

**Comment by gucio321** [comment]
is there anyone from brave on this repo? This issue is open since october and no maintainer's investigation yet...

@jorgebotas try to close brave completly that worked for me

**Comment by MJSBDLK** [comment]
Got this bug as well, and was able to fix it by closing each tab individually.

Ubuntu 24.04.1 LTS
GNOME Shell 46.0
XDG_SESSION_TYPE x11
brave  1.73.104  464  latest/stable
[Version 1.73.104 Chromium: 131.0.6778.204 (Official Build) (64-bit)]

**Comment by ezekias42** [comment]
This did work for me, is there any way this becomes the default on new Brave releases?

**Comment by piglesias1** [comment]
@ezekias42, not sure if this is related to Brave, as it was reported to happen with other browsers too. Probably a shortcut (disabled by default) or something else in Gnome. I have different profiles and happen only with some of them. Also, it was reported to solve the issue after closing all the tabs, which resets the visible workspace function when a new window is open. That's only my guess. Maybe Brave team can comment something about it.

**Comment by waka-man** [comment]
> Not really an answer for Brave, but I had the same issue with Vivaldi in Gnome and [@piglesias1](https://github.com/piglesias1) suggestion helped. Browser window had the "Always on Visible Workspace" checked by default, but the browser does not display the system title bar, I had to go to `Settings -> Appearance -> Window Appearance` and select "Use Native Window" option (which required a restart of the browser). Right-clicking on the system title bar allows you to un-check the option to keep the browser always on visible workspace.

This is what ended up working without doing all the other suggested acrobatics

**Comment by jgtvares** [comment]
A simple reboot did the trick for me.

**Comment by trbntwo** [comment]
> Got this bug as well, and was able to fix it by closing each tab individually.
> 
> Ubuntu 24.04.1 LTS GNOME Shell 46.0 XDG_SESSION_TYPE x11 brave 1.73.104 464 latest/stable [Version 1.73.104 Chromium: 131.0.6778.204 (Official Build) (64-bit)]

This fixed for me as well, but I guess this means this is not a GNOME bug or feature and there is indeed a bug in Brave
